actionview/lib/action_view/template/resolver.rb

```diff
@@ -217,11 +217,9 @@ def eql?(resolver)
     alias :== :eql?
 
     def all_template_paths # :nodoc:
-      paths = Dir.glob("**/*", base: @path)
-      paths.reject do |filename|
-        File.directory?(File.join(@path, filename))
-      end.map do |filename|
-        filename.gsub(/\.[^\/]*\z/, "")
+      paths = template_glob("**/*")
+      paths.map do |filename|
+        filename.from(@path.size + 1).gsub(/\.[^\/]*\z/, "")
```

Comment:
I find the `+ 1` a little distracting from the flow, do you think it would be clearer if we used `Integer#next` here?

```suggestion
        filename.from(@path.size.next).gsub(/\.[^\/]*\z/, "")
```

Reply:
Since we're using Active Support's `String#from` extension, we could use `remove` too:

```suggestion
        filename.from(@path.size + 1).remove(/\.[^\/]*\z/)
```